### PR TITLE
Restrict cipher suites for Desktop Access in FIPS mode

### PR DESCRIPTION
**Pull request by ibeckermayer**
This PR changes the boring hash in the Cargo.toml to one which allows us to call `builder.set_fips_compliance_policy()?;`, which restricts cipher suites to those which are FIPS validated. To this end, it also reverts clang from 14 to 12.0.0 in the buildbox, because the boring hash we're now using is using a (actually, not soon-to-be) FIPS-validated hash for boringssl which must be built by 12.0.0.

It also removes the RDP client (and therefore Desktop Access feature) for FIPS builds on arm64. It was proving extremely difficult to get boringssl to build via the Centos 7 Docker image. Since we are unaware of paying customers who require running a FIPS-enabled Windows Desktop Service on this architecture, we've elected to just drop support for now.

Closes https://github.com/gravitational/teleport.e/issues/3065
Closes https://github.com/gravitational/customer-sensitive-requests/issues/254

Depends on
- https://github.com/gravitational/boring/pull/9
- https://github.com/gravitational/teleport.e/pull/4299

### CODE REVIEW

File: build.assets/Dockerfile-centos7-assets [hunk 1]
```diff
@@ -60,21 +60,18 @@ RUN git clone --depth=1 https://github.com/Kitware/CMake.git -b v3.28.1 && \
...
-        scl enable ${DEVTOOLSET} "cmake -Bbuild-cmake && \
+    cd ninja && \
+    [ "$(git rev-parse HEAD)" = 'a524bf3f6bacd1b4ad85d719eed2737d8562f27a' ] && \
+    scl enable ${DEVTOOLSET} "cmake -Bbuild-cmake && \
     cmake --build build-cmake -j"$(nproc)" && \
     cmake --build build-cmake --target  install"
 
 # Use just created devtool image with newer GCC and Cmake
-FROM --platform=$BUILDPLATFORM centos-devtoolset as clang14
+FROM --platform=$BUILDPLATFORM centos-devtoolset as clang12
```
Comment (probakowski) [inline build.assets/Dockerfile-centos7-assets:70]: Is this to revert back to FIPS-validated build of boring?

**Review by probakowski — APPROVED** [review]
LGTM, just one question

File: build.assets/Dockerfile-centos7-assets (hunk 1 above)
Comment (ibeckermayer) [reply build.assets/Dockerfile-centos7-assets:70]: Yes, that should be in the description, thanks.

File: Makefile [hunk 2]
```diff
@@ -112,14 +112,23 @@ ifeq ($(RDPCLIENT_SKIP_BUILD),0)
...
 
-# Do not build RDP client on ARM or 386.
+is_fips_on_arm64 := no
+ifneq ("$(FIPS)","")
+ifeq ("$(ARCH)","arm64")
+is_fips_on_arm64 := yes
+endif
+endif
+
+# Do not build RDP client on ARM or 386, or for FIPS builds on arm64.
```
Comment (zmb3) [inline Makefile:122]:
```suggestion
# Do not build RDP client on 32-bit ARM or 386, or for FIPS builds on arm64.
```

**Review by zmb3 — APPROVED** [review]

**Comment by ibeckermayer** [comment]
I've added this note to the documentation re: Desktop Access FIPS: ab64baa17a5e3b8c604275b0c6f192d669634e66

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 3]
```diff
@@ -282,3 +282,11 @@ is emitted to the Audit Log.
 - All uses of non-compliant algorithms such as NaCl are removed and replaced with compliant algorithms such as AES-GCM.
 - Teleport is compiled with [BoringCrypto](https://csrc.nist.gov/projects/cryptographic-module-validation-program/certificate/4407).
 - User, host, and CA certificates (and host keys for recording proxy mode) only use 2048-bit RSA private keys.
+
+## Desktop Access
```
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:286]:
```suggestion
## Remote desktop access
```

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 4]
```diff
@@ -282,3 +282,11 @@ is emitted to the Audit Log.
...
+
+Teleport's Desktop Access feature uses Rust for RDP connections, and thus uses a fork of Cloudflare's [`boring`](https://github.com/gravitational/boring)
```
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:288]:
```suggestion
Teleport uses Rust for RDP connections, and thus uses a fork of Cloudflare's [`boring`](https://github.com/gravitational/boring)
```

File: docs/pages/access-controls/compliance-frameworks/fedramp.mdx [hunk 5]
```diff
@@ -282,3 +282,11 @@ is emitted to the Audit Log.
...
+library under the hood for FIPS-compliant TLS cryptography. The primary noteable difference to the specifications listed above is that
+TLS is restricted to TLS 1.2 only (1.3 is not supported).
+
+Note that Teleport's Desktop Access feature is not available in FIPS mode on `arm64`.
```
Comment (zmb3) [inline docs/pages/access-controls/compliance-frameworks/fedramp.mdx:292]:
```suggestion
Note that `arm64` FIPS builds do not support access to Windows desktops.
```

**Pull request merged**